feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/TimetableGridItem.kt

```diff
@@ -108,41 +109,67 @@ fun TimetableGridItem(
             }
             .padding(TimetableGridItemSizes.padding),
     ) {
-        Column {
+        val (titleRef, timeRef, speakerRef) = createRefs()
+        Text(
+            modifier = Modifier.constrainAs(titleRef) {
+                top.linkTo(parent.top)
+                start.linkTo(parent.start)
+                bottom.linkTo(timeRef.top)
+            }.heightIn(0.dp, height.div(2.5f)),
+            text = timetableItem.title.currentLangTitle,
+            style = titleTextStyle,
+            overflow = TextOverflow.Ellipsis,
+        )
+
+        Row(
+            modifier = Modifier
+                .padding(top = TimetableGridItemSizes.titleToSchedulePadding)
+                .constrainAs(timeRef) {
+                    start.linkTo(parent.start)
+                    top.linkTo(titleRef.bottom)
+                    bottom.linkTo(speakerRef.top)
+                    this.height = Dimension.fillToConstraints
+                },
+        ) {
+            Icon(
+                modifier = Modifier.height(TimetableGridItemSizes.scheduleHeight),
+                imageVector = Icons.Default.Schedule,
+                tint = if (speaker != null) {
+                    hallColor.hallText
+                } else {
+                    hallColor.hallTextWhenWithoutSpeakers
+                },
+                contentDescription = ScheduleIcon.asString(),
+            )
+            Spacer(modifier = Modifier.width(4.dp))
             Text(
-                text = timetableItem.title.currentLangTitle,
-                style = titleTextStyle,
+                modifier = Modifier.height(TimetableGridItemSizes.scheduleHeight),
+                text = "${timetableItem.startsTimeString} - ${timetableItem.endsTimeString}",
+                style = MaterialTheme.typography.bodySmall,
+                color = textColor,
             )
-            Spacer(modifier = Modifier.height(TimetableGridItemSizes.titleToScheduleSpaceHeight))
-            Row(modifier = Modifier.height(TimetableGridItemSizes.scheduleHeight)) {
-                Icon(
-                    imageVector = Icons.Default.Schedule,
-                    tint = if (speaker != null) {
-                        hallColor.hallText
-                    } else {
-                        hallColor.hallTextWhenWithoutSpeakers
-                    },
-                    contentDescription = ScheduleIcon.asString(),
-                )
-                Spacer(modifier = Modifier.width(4.dp))
-                Text(
-                    text = "${timetableItem.startsTimeString} - ${timetableItem.endsTimeString}",
-                    style = MaterialTheme.typography.bodySmall,
-                    color = textColor,
-                )
-            }
+        }
 
-            Spacer(
-                modifier = Modifier
-                    .weight(1f)
-                    .defaultMinSize(minHeight = 8.dp),
-            )
+        if (speakers.isEmpty()) return@ConstraintLayout
```

Comment:
Almost LGTM! I would like to use if block here. This is because we don't expect an early return when we write Compose so we could accidentally skip important component 🙏

Reply:
Added.